### PR TITLE
Recognize Numeric Tags

### PR DESCRIPTION
**Pull request by njnygaard**
Numeric tags not being recognized in tag explorer #451

I was able to get up and running and I believe that this solves the issue.
Please let me know if I need to make any modifications.

I also added the debug configuration because I needed it while working.

### CODE REVIEW

**Comment by njnygaard** [comment]
```
FAIL test/utils.test.ts
  ● hashtag extraction › ignores tags that only have numbers in text

    expect(received).toEqual(expected) // deep equality

    - Expected
    + Received

      Set {
    +   "123",
        "123four",
      }

      94 |     expect(
      95 |       extractHashtags('this #123 tag should be ignore, but not #123four')
    > 96 |     ).toEqual(new Set(['123four']));
         |       ^
      97 |   });
      98 | 
      99 |   it('ignores hashes in plain text urls and links', () => {

      at Object.<anonymous> (test/utils.test.ts:96:7)
```
Interesting, I thought the point was to allow numeric tagging? This test seems antithetical to the goal of #451.
I'm hesitant to remove a test without feedback.

**Review by riccardoferretti — COMMENTED** [review]
To answer your question around the tests, originally we didn't want to support numeric tags, but changed direction later - apparently this slipped past review.

The more I think about it the more I don't love numeric tags simply being included, if we want to enable them it should be via a setting (cc @jmg-duarte @ingalless who have been involved in previous tag related conversations)

File: .vscode/launch.json [hunk 1]
```diff
@@ -44,6 +44,18 @@
 			],
 			"preLaunchTask": "${defaultBuildTask}"
 		},
+		{
+			"name": "Debug VSCode Extension",
+			"type": "extensionHost",
+			"request": "launch",
+			"runtimeExecutable": "${execPath}",
+			"args": [
+				"--inspect-brk",
```
[the remark this answers is not in the document]
Comment (njnygaard) [reply .vscode/launch.json:53]: This is my first time working with VSCode extensions. To your point, I don't think the debug configuration is needed. With a clearer head this morning, I was able to get my `debugger;` working with the original run configuration. I've removed the debug configuration from this PR.

**Comment by njnygaard** [comment]
I can try to work on encapsulating the numeric tag behavior in a setting like `foam.numericTags`. Hopefully that's not too far out of scope of "good first issue".

**Comment by SanketDG** [comment]
> I can try to work on encapsulating the numeric tag behavior in a setting like `foam.numericTags`. Hopefully that's not too far out of scope of "good first issue".

Sounds good to me.

I would have never have thought of the original use-case in the issue, that's pretty legitimate, so I would also be okay if it's supported out of the box.

**Comment by riccardoferretti** [comment]
> I can try to work on encapsulating the numeric tag behavior in a setting like foam.numericTags. Hopefully that's not too far out of scope of "good first issue".

Sounds good to me, minor detail: I would call it `foam.tags.includeNumbers` and let's default it to `false` (as that seems to be the standard behaviour in various places, from Twitter to Obsidian)

**Pull request closed without merging**